addons/sale/models/sale.py

```diff
@@ -402,6 +403,28 @@ def _onchange_commitment_date(self):
                 }
             }
 
+    @api.onchange('pricelist_id', 'order_line')
+    def _onchange_pricelist_id(self):
+        if self.order_line and self.pricelist_id and self._origin.pricelist_id and self._origin.pricelist_id != self.pricelist_id:
+            self.is_change_pricelist = True
+        else:
+            self.is_change_pricelist = False
+
+    def update_pricelist(self):
+        self.ensure_one()
+        for line in self.order_line.filtered(lambda line: not line.display_type):
+            product = line.product_id.with_context(
+                partner=self.partner_id,
+                quantity=line.product_uom_qty,
+                date=self.date_order,
+                pricelist=self.pricelist_id.id,
+                uom=line.product_uom.id
+            )
+            line.price_unit = self.env['account.tax']._fix_tax_included_price_company(
+                line._get_display_price(product), line.product_id.taxes_id, line.tax_id, line.company_id)
+        self.is_change_pricelist = False
+        self.message_post(body=_("Updated product price according to pricelist <b>%s<b> ") % self.pricelist_id.display_name)
```

Comment:
`"The prices have been recomputed according to pricelist  <b>%s<b> "`

Reply:
@Feyensv Done.